projectile.el

```diff
@@ -1946,14 +1946,17 @@ It assumes the test/ folder is at the same level as src/."
       (let ((impl-file (projectile-find-matching-file file-name)))
         (if impl-file
             (projectile-expand-root impl-file)
-          (error "No matching source file found")))
+          (error
+           (format "No matching source file found for project type: %s"
+                   (projectile-project-type)))))
```

Comment:
Or we can simply display the project type at the modeline, which seems like a better idea to me.